src/node_dns.cc

```diff
@@ -0,0 +1,471 @@
+
+  std::vector<getdns_transport_list_t> transport_list(length);
+  for (uint32_t i = 0; i < length; i += 1) {
+    getdns_transport_list_t transport =
+        static_cast<getdns_transport_list_t>(transports->Get(env->context(), i)
+                                                 .ToLocalChecked()
+                                                 ->Uint32Value(env->context())
+                                                 .ToChecked());
+    transport_list[i] = transport;
+  }
+
+  GETDNS(context_set_dns_transport_list,
+         env->isolate(),
+         dns->context_,
+         length,
+         &transport_list[0]);
+}
+
+// static
+void DNSWrap::CancelAllTransactions(const FunctionCallbackInfo<Value>& args) {
+  DNSWrap* dns;
+  ASSIGN_OR_RETURN_UNWRAP(&dns, args.This());
+  std::vector<getdns_transaction_t> transactions;
+  for (auto& t : dns->transactions_) {
+    transactions.push_back(t.first);
+  }
+  for (auto& t : transactions) {
+    GETDNS(cancel_callback, dns->env()->isolate(), dns->context_, t);
+  }
+  dns->transactions_.clear();
+}
+
+static void Initialize(Local<Object> target,
+                       Local<Value> unused,
+                       Local<Context> context,
+                       void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+
+  Local<FunctionTemplate> tmpl = env->NewFunctionTemplate(DNSWrap::New);
+  auto dns_wrap_string = FIXED_ONE_BYTE_STRING(env->isolate(), "DNSWrap");
+  tmpl->InstanceTemplate()->SetInternalFieldCount(DNSWrap::kInternalFieldCount);
+  tmpl->SetClassName(dns_wrap_string);
+
+  env->SetProtoMethod(tmpl, "getAddresses", DNSWrap::GetAddresses);
+  env->SetProtoMethod(tmpl, "getServices", DNSWrap::GetServices);
+  env->SetProtoMethod(tmpl, "getHostnames", DNSWrap::GetHostnames);
+  env->SetProtoMethod(tmpl, "getGeneral", DNSWrap::GetGeneral);
+
+  env->SetProtoMethod(tmpl,
+                      "setUpstreamRecursiveServers",
+                      DNSWrap::SetUpstreamRecursiveServers);
+  env->SetProtoMethod(tmpl,
+                      "getUpstreamRecursiveServers",
+                      DNSWrap::GetUpstreamRecursiveServers);
+  env->SetProtoMethod(
+      tmpl, "setDNSTransportList", DNSWrap::SetDNSTransportList);
+
+  env->SetProtoMethod(
+      tmpl, "cancelAllTransactions", DNSWrap::CancelAllTransactions);
+
+  target
+      ->Set(env->context(),
+            dns_wrap_string,
+            tmpl->GetFunction(context).ToLocalChecked())
+      .ToChecked();
+
+#define V(name)                                                                \
+  target                                                                       \
+      ->Set(env->context(),                                                    \
+            FIXED_ONE_BYTE_STRING(env->isolate(), #name),                      \
+            Integer::New(env->isolate(), name))                                \
+      .ToChecked();
+
+  V(GETDNS_TRANSPORT_UDP)
+  V(GETDNS_TRANSPORT_TCP)
+  V(GETDNS_TRANSPORT_TLS)
```

Comment:
Are these constants? If so, Why not use the existing macros for setting constants?

Reply:
What are the existing macros for this?